mock-data/generateData.js

```diff
@@ -1,120 +1,155 @@
       available = false;
-      prevBookingDuration = chance.integer({ min: MIN_STAY_DEFAULT, max: MAX_STAY_DEFAULT });
-      mockBookings.push(createBooking(day, prevBookingDuration, listingID, users));
+      prevBookingDuration = generateInteger(MAX_STAY_DEFAULT, MIN_STAY_DEFAULT);
+      array.push(createBooking(listingId, day, prevBookingDuration));
     }
     daysAway += 1;
     reduced = Math.floor(daysAway * 0.25);
   }
-  return mockBookings;
+  return array.join('');
 };
 
-const generateAllBookings = (listings = LISTINGS_DEFAULT, users = USERS_DEFAULT) => {
-  const today = moment().format('YYYY-MM-DD');
-  const periodEnd = moment(today).clone().add(2, 'years').format('YYYY-MM-DD');
-  const bookingRange = moment.range(today, periodEnd);
-  const allListingBookings = [];
-
-  for (let i = 1; i <= listings; i += 1) {
-    const listingID = i;
-    allListingBookings.push(addMockBookings(listingID, bookingRange, users));
+const generateAllBookings =  (listingStart, listingEnd, callback) => {
+  for (let i = listingStart; i <= listingEnd; i += 1) {
+    addMockBookings(i, new Date(), 90, callback);
   }
-
-  return allListingBookings.reduce((a, b) => a.concat(b), []);
 };
 
-const generateFirstName = (users = OWNERS_DEFAULT) => {
-  const accounts = [];
-  for (let i = 1; i <= users; i += 1) {
-    const user = chance.first();
-    let double = "";
-    user.split('').forEach((char) => {
-      double += char;
-    });
-    accounts.push(double);
+const generateOneBooking = (listingId) => addMockBookings(listingId, START_DATE, 90);
+
+const generateFirstName =  (listingStart, listingEnd, callback) => {
+  const userRatio = USERS_DEFAULT / LISTINGS_DEFAULT;
+  for (let i = (listingStart - 1) * userRatio + 1 ; i <= listingEnd * userRatio; i++) {
+    if (callback)  callback(`${chance.first()}\n`);
   }
-  return accounts;
 };
 
-const createReview = (listingID) => {
-  const review = [];
-  const rating = chance.integer({ min: 0, max: 5 });
-  review.push(rating);
-  review.push(listingID);
-  return review;
-};
+const generateOneFirstName = () => {
+    return `${chance.first()}\n`;
+}
+
+const createReview = (listingID) => (
+  `${
+    generateInteger(5, 2)
+  },${
+    listingID
+  }\n`
+);
 
-const generateAllReviews = (listings = LISTINGS_DEFAULT, maxReviews = MAX_REVIEWS_DEFAULT) => {
-  const allListingReviews = [];
-  for (let i = 1; i <= listings; i += 1) {
-    const numOfListingReviews = chance.integer({ min: 1, max: maxReviews });
-    console.log(numOfListingReviews);
-    const currentListingID = i;
-    for (let j = 0; j < numOfListingReviews; j += 1) {
-      allListingReviews.push(createReview(currentListingID));
+const generateAllReviews =  (listingStart, listingEnd, callback) => {
+  for (let i = listingStart; i <= listingEnd; i++) {
+    const numOfListingReviews = generateInteger(MAX_REVIEWS_DEFAULT, 1);
+    for (let j = 0; j < numOfListingReviews; j++) {
+      if (callback)  callback(createReview(i));
```

Comment:
curly brackets around callback(createReview(i)) in if statements for best practice.

Reply:
True, I will incorporate that in the next pr